toolchain/driver/driver_subcommand.h

```diff
@@ -32,16 +32,31 @@ class DriverSubcommand {
   // the subcommand is in use.
   auto AddTo(CommandLine::CommandBuilder& b,
              DriverSubcommand** selected_subcommand) -> void {
-    b.AddSubcommand(info_, [this, selected_subcommand](
-                               CommandLine::CommandBuilder& sub_b) {
-      BuildOptions(sub_b);
-      sub_b.Do([this, selected_subcommand] { *selected_subcommand = this; });
-    });
+    b.AddSubcommand(
+        info_, [this, selected_subcommand](CommandLine::CommandBuilder& sub_b) {
+          BuildOptionsAndSetAction(sub_b, selected_subcommand);
+        });
   }
 
   // Adds command line options.
   virtual auto BuildOptions(CommandLine::CommandBuilder& b) -> void = 0;
 
+  // Adds command line options and registers the `Run` method to be called.
```

Comment:
Can we make this and `BuildOptions` private, to make it clearer that they are following the NVI pattern?

Reply:
I've already left a TODO to revisit how this is structured. I'd rather not expand the scope of the change to move both, and it felt weird to make this one different from the above.